frontend/src/components/Notes.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { useState, useEffect } from "react"
+import { formatDistanceToNow } from "date-fns"
+import { Avatar } from "@/components/Avatar"
+import {
+  IRecipe,
+  addNoteToRecipe,
+  toggleCreatingNewNote,
+  setDraftNote,
+  updateNote,
+  toggleEditingNoteById,
+  deleteNote
+} from "@/store/reducers/recipes"
+import { ButtonPrimary, ButtonSecondary } from "@/components/Buttons"
+import { classNames } from "@/classnames"
+import { useDispatch, useSelector } from "@/hooks"
+import { isSuccessLike } from "@/webdata"
+import orderBy from "lodash/orderBy"
+import Textarea from "react-textarea-autosize"
+import { Markdown } from "@/components/Markdown"
+
+interface INoteProps {
+  readonly note: IRecipe["notes"][0]
+  readonly recipeId: IRecipe["id"]
+  readonly className?: string
+}
+export function Note({ recipeId, note, className }: INoteProps) {
+  const dispatch = useDispatch()
+  const [newText, setNewText] = useState(note.text)
+  useEffect(() => {
+    setNewText(note.text)
+  }, [note])
+  const created_by = note.created_by
+  const isOpen = useSelector(s => {
+    const noteData = s.recipes.notesById[note.id]
+    if (!noteData) {
+      return false
+    }
+    return Boolean(noteData.openForEditing)
+  })
+  const updating = useSelector(s => {
+    const noteData = s.recipes.notesById[note.id]
+    if (!noteData) {
+      return false
+    }
+    return Boolean(noteData.saving)
+  })
+
+  const onCreate = () => {
```

Comment:
```suggestion
  const onUpdate = () => {
```

It might be nicer to pull these out into a hook

e.g. `useNoteEditHandler()`